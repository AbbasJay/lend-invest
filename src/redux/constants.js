const prefix = "currentLoans/";

export const SET_CURRENT_LOANS = `${prefix}SET_CURRENT_LOANS`;

export const SET_SELECTED_LOAN = `${prefix}SET_SELECTED_LOAN`;

export const SHOW_INVEST_MODAL = `${prefix}SHOW_INVEST_MODAL`;
export const HIDE_INVEST_MODAL = `${prefix}HIDE_INVEST_MODAL`;

export const SET_INVESTED_AMOUNT = `${prefix}SET_INVESTED_AMOUNT`;
