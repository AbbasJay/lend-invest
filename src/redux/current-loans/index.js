const currentLoansReducer = (state = []) => {
  return state;
};

export default currentLoansReducer;
