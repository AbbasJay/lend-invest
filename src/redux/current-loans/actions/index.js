import * as constants from "../constants";

export const setCurrentLoans = (currentLoans) => ({
  type: constants.SET_CURRENT_LOANS,
  payload: { currentLoans },
});
