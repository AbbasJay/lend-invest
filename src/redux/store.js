import { createStore, combineReducers } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";

import { currentLoansReducer } from "./reducers/currentLoans";
import { selectedLoanReducer } from "./reducers/selectedLoan";
import { uiReducer } from "./reducers/ui";

const store = createStore(
  combineReducers({
    currentLoans: currentLoansReducer,
    selectedLoan: selectedLoanReducer,
    ui: uiReducer,
  }),
  composeWithDevTools()
);

export default store;
