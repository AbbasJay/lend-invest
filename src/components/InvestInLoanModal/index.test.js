import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";

import { mount } from "enzyme";

import { InvestInModal } from "./";

import store from "../../redux/store";

import { showInvestModal, setSelectedLoan } from "../../redux/actions";

const selectedLoan = {
  title: "title 1.",
  amount: "85,754",
  id: "1",
  available: "8,250",
  annualised_return: "1.5",
  ltv: "48.80",
  amount: "85,754",
};

const renderedComponent = () => {
  return mount(
    <Provider store={store}>
      <InvestInModal />
    </Provider>
  );
};

describe("InvestInModal", () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it("renders correctly to snapshot", () => {
    store.dispatch(setSelectedLoan(selectedLoan));
    store.dispatch(showInvestModal());

    const component = renderedComponent();
    const snapshot = renderer.create(component).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it("should render component", () => {
    const component = renderedComponent();

    expect(component.length).toBe(1);
  });
});
