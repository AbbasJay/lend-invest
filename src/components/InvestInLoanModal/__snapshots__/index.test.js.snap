// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestInModal renders correctly to snapshot 1`] = `
<div
  className="invest-modal-background"
  onClick={[Function]}
>
  <div
    className="invest-modal-container"
  />
  <div
    className="invest-modal-content"
  >
    <h1>
      Invest in loan
    </h1>
    <div
      className="invest-details"
    >
      <div
        className="invest-title"
      >
        <span>
          title 1.
        </span>
      </div>
      <div
        className="invest-items"
      >
        <span>
          Amount available: £8,250
        </span>
        <span>
          Loan ends in: NaN days
        </span>
      </div>
      <div
        className="invest-amount-text"
      >
        <span>
          Investment amount (£)
        </span>
      </div>
    </div>
    <div
      className="invest-amount-container"
    >
      <input
        className="invest-input"
        onChange={[Function]}
      />
      <button
        className="invest-button"
        onClick={[Function]}
      >
        INVEST
      </button>
    </div>
    
  </div>
</div>
`;
